enhancements/network/egressip_for_ovn_kubernetes.md

```diff
@@ -0,0 +1,148 @@
+assign. For automatic assignment to be done, the user needs to set `EgressCIDRs` or multiple `EgressIPs` on the `hostsubnet` object, else: 
+the assignment is left up to the user's own management. Egress IP is managed in a highly available manner if automatic assignment 
+is possible, meaning: if the node currently hosting that IP falls off the cluster, the network plugin re-assigns the egress IP to another
+cluster node as a fallback mechanism.
+
+If we summarize what openshift-sdn concretely achieves by its two-step process, we have the following: 
+
+1.1. It allows the cluster admin to have a direct say on which nodes act as egress nodes.
+1.2. The assignment of egress IPs to namespaces is a manual process (defined by the cluster admin). The assignment of egress IPs to 
+nodes is performed automatically by the network plugin. In the case of a node failure: the egress IPs hosted on that node are 
+re-distributed to another node in the cluster which can host them.
+
+All in all, the feature is quite complex as it introduces a lot of notions which obfuscate these two final functionalities/goals.  
+
+This enhancement proposal focuses on how this feature will work with OVN-Kubernetes.
+
+## Motivation
+
+The primary motivation is making OVN-Kubernetes have feature parity with openshift-sdn. Secondary motivations come partially from 
+product management: enhancing the feature by allowing cluster admins the freedom to control cluster egress traffic 
+on pod level, and partially from us in the networking team; simplifying the usability for cluster admins and our 
+maintenance of the code base which implements it.   
+
+### Goals & Requirements
+
+2.1. The egress IP functionality for OVN-Kubernetes needs to have feature parity with openshift-sdn, according to 1.1. and 1.2.
+2.2. The egress IP functionality for OVN-Kubernetes should allow pod selection within a namespace when defining the IP of egress traffic. 
+2.3. The egress IP functionality for OVN-Kubernetes should support single-stack IPv4/IPv6 as well as dual-stack, in line with global 
+OpenShift requirements for all networking functionality for release 4.6.
+
+This proposal details how OVN-Kubernetes goes about addressing these goals in the section [Proposal](#proposal)
+ 
+### Non-Goals
+
+3.1. Making an EgressIP object cluster-wide, it will still be namespaced
+
+## Proposal
+
+We propose to make the assignment of egress IPs simpler by defining the following sequence of events for a user wishing to define an egress 
+IP for its cluster.
+
+As to satisfy 1.1. we propose that cluster admins tag the nodes they wish to use as egress nodes by labelling them with 
+`k8s.ovn.org/egress-assignable: ""`. In case an `EgressIP` object is created and no nodes are tagged as such, the egress IP creation is 
+invalidated. To rectify that: a cluster admin would have to; delete the previously created `EgressIP` object, 
+label the nodes needed, and re-create the previous object. 
```

Comment:
They should not need to delete the `EgressIP` object. Just labelling the nodes should fix it

Reply:
I am not a fan of that because it adds unwanted complexity and I suspect there might be confusion from a user's point of view when that _exact_ condition does not apply, say: if they label one node, then start creating `EgressIP`s which all get assigned to that one node, and then label a few more nodes. The old `EgressIP`s will not be re-assigned, but they might expect it to be.

Reply:
Well, finally I implemented it. So let's keep that. I updated the enhancement proposal to reference that any `EgressIP` object created while there were assignable nodes will not be updated.